bin/src/hickory-dns.rs

```diff
@@ -240,37 +242,33 @@ async fn async_run(args: Cli) -> Result<(), String> {
         .unwrap_or(directory_config);
 
     #[cfg(feature = "prometheus-metrics")]
-    if !args.disable_prometheus && !config.disable_prometheus() {
-        let mut socket = args
+    let prometheus_server_handles = if !args.disable_prometheus && !config.disable_prometheus() {
+        let socket_addr = args
```

Comment:
If we're going to have a separate `prometheus_server` module, I'd try to move as much code as possible in there, and expose an API that is as small as possible.

Reply:
I encapsulated more of this in a struct. I'm inclined to leave the socket setup here, at least for now, because it's coupled to the configuration struct, the other "listening for _ on _" log messages (that integration tests rely on) are here as well, and `build_tcp_listener()` is defined in this file.